tests/mypy/test_mypy.py

```diff
@@ -35,52 +48,44 @@
 os.chdir(Path(__file__).parent.parent.parent)
 
 
-@dataclasses.dataclass
-class MypyCasesBuilder:
-    configs: Union[str, List[str]]
-    modules: Union[str, List[str]]
-    marks: Any = None
-
-    def build(self) -> List[Union[Tuple[str, str], Any]]:
-        """
-        Produces the cartesian product of the configs and modules, optionally with marks.
-        """
-        if isinstance(self.configs, str):
-            self.configs = [self.configs]
-        if isinstance(self.modules, str):
-            self.modules = [self.modules]
-        built_cases = []
-        for config in self.configs:
-            for module in self.modules:
-                built_cases.append((config, module))
-        if self.marks is not None:
-            built_cases = [pytest.param(config, module, marks=self.marks) for config, module in built_cases]
-        return built_cases
-
-
-cases = (
+# Type hint taken from the signature of `pytest.param`:
+Marks: TypeAlias = MarkDecorator | Collection[MarkDecorator | Mark]
+
+
+def build_cases(
```

Comment:
Replaces `MypyCasesBuilder` (no need for a class here) and also enforce providing `list[str]` for the arguments, which simplifies the body to a single list comprehension.

Reply:
Great simplification!